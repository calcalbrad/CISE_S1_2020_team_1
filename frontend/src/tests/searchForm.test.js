import React from 'react';
import ReactDOM from 'react-dom';
import SearchForm from '../components/searchForm'
import Enzyme, { render, ReactWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme'
import { getByTestId, fireEvent } from '@testing-library/react';
Enzyme.configure({ adapter: new Adapter()})

describe('Search Form', () => {

  it('renders without crashing', () => {
    const component = mount(<MessageForm/>);
    expect(component).toMatchSnapshot();
  });

  it('has textbox', () => {
    const component = mount(<MessageForm/>);
    expect(component.exists('textarea#message_box'));
  });

  it('has submit button', () => {
    const component = mount(<MessageForm/>);
    expect(component.exists('button#submit'));
  });

  it('should update state message when text entered', () => {
    //const component = shallow(<MessageForm/>);
    //component.find('textarea#message_box').simulate('change', { target: { value: 'Hello' } });
    //expect(component.state('currentMessage')).toEqual('Hello');

    const container = mount(<MessageForm/>);
    const text_box = container.find('message_box')
    text_box.simulate('change', {target: { value: 'Hello' } });
    expect(component.state('currentMessage').toEqual('Hello'));
  });

  it('clears message box on submit', () => {
    const component = mount(<MessageForm
      submitMessage={function(item){return true}}
    />);
    component.find('textarea#message_box').simulate('change', { target: { value: 'Hello' } })
    expect(component.state('currentMessage')).toEqual('Hello');
    component.find('form').simulate('submit')
    component.update()
    expect(component.find('textarea#message_box').props().value).toEqual('');
    expect(component.state('currentMessage')).toEqual('');
  });

});
