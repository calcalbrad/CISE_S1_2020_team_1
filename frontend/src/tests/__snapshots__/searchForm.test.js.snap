// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form renders without crashing 1`] = `ReactWrapper {}`;
